includes/Checker/Runtime_Environment_Setup.php

```diff
@@ -110,6 +110,14 @@ public function clean_up() {
 		$prefix_cleanup = $this->amend_db_base_prefix();
 		$tables         = $wpdb->tables();
 
+		// Remove tables not starting with the prefix.
+		$tables = array_filter(
+			$tables,
+			function ( $table ) use ( $wpdb ) {
+				return str_starts_with( $table, $wpdb->prefix );
+			}
+		);
```

Comment:
How about this instead? It focuses more concretely on the problem and is more efficient than iterating through all tables.
```suggestion
		// Do not remove custom tables (which by definition weren't duplicated because we cannot override constants).
		if ( isset( $tables['users'] ) && defined( 'CUSTOM_USER_TABLE' ) && $tables['users'] === CUSTOM_USER_TABLE ) {
			unset( $tables['users'] );
		}
		if ( isset( $tables['usermeta'] ) && defined( 'CUSTOM_USER_META_TABLE' ) && $tables['usermeta'] === CUSTOM_USER_META_TABLE ) {
			unset( $tables['usermeta'] );
		}
```